packages/cli/src/UserManagement/email/UserManagementMailer.ts

```diff
@@ -22,58 +21,37 @@ import { EventRelay } from '@/eventbus/event-relay.service';
 type Template = HandlebarsTemplateDelegate<unknown>;
 type TemplateName = 'invite' | 'passwordReset' | 'workflowShared' | 'credentialsShared';
 
-const templates: Partial<Record<TemplateName, Template>> = {};
-
-async function getTemplate(
-	templateName: TemplateName,
-	defaultFilename = `${templateName}.html`,
-): Promise<Template> {
-	let template = templates[templateName];
-	if (!template) {
-		const templateOverride = config.getEnv(`userManagement.emails.templates.${templateName}`);
-
-		let markup;
-		if (templateOverride && existsSync(templateOverride)) {
-			markup = await readFile(templateOverride, 'utf-8');
-		} else {
-			markup = await readFile(pathJoin(__dirname, `templates/${defaultFilename}`), 'utf-8');
-		}
-		template = Handlebars.compile(markup);
-		templates[templateName] = template;
-	}
-	return template;
-}
-
 @Service()
 export class UserManagementMailer {
 	readonly isEmailSetUp: boolean;
 
-	private mailer: NodeMailer | undefined;
+	readonly templateOverrides: GlobalConfig['userManagement']['emails']['template'];
+
+	readonly templatesCache: Partial<Record<TemplateName, Template>> = {};
+
+	readonly mailer: NodeMailer | undefined;
 
 	constructor(
-		private readonly userRepository: UserRepository,
+		globalConfig: GlobalConfig,
 		private readonly logger: Logger,
+		private readonly userRepository: UserRepository,
 		private readonly urlService: UrlService,
 	) {
-		this.isEmailSetUp =
-			config.getEnv('userManagement.emails.mode') === 'smtp' &&
-			config.getEnv('userManagement.emails.smtp.host') !== '';
+		const emailsConfig = globalConfig.userManagement.emails;
+		this.isEmailSetUp = emailsConfig.mode === 'smtp' && emailsConfig.smtp.host !== '';
```

Comment:
We could consider 'getters' for the new config.

Reply:
That is also a part of the plan, to have more utility methods/getters on the config classes.